#VRML_SIM R2021b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.5013444099918104 -0.6921234192420614 -0.5192484521947477 4.367963173968041
  position 0.5720016504945044 2.1275975394671995 0.049330455987781234
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 0.254 1.524
  floorTileSize 0.508 0.508
}
DEF epuck E-puck {
  translation 0 0 0.508
  rotation -0.008662457471447753 -0.9999624798217843 2.7914564127265318e-05 0.52
  controller "lab2_task1"
  supervisor TRUE
  camera_width 50
  turretSlot [
    InertialUnit {
      rotation 0 1 0 3.14
    }
    DEF righthinge HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
          controlPID 8 2 0.5
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.04500995778380096 0.025013380758883635 -2.4671929852070184e-05
        rotation 0.9999993203499804 0.0010052734270078164 -0.0005905293508524926 4.6368371220546685
        children [
          DEF WHEEL Transform {
            translation -0.007 0.01 -0.05
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                  transparency 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
            rotationStep 0.25
          }
        ]
        name "right wheel(1)"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
        rotationStep 0.25
      }
    }
    DEF lefthinge HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
          controlPID 8 0 0
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.04501012942703836 0.024977671977279067 -4.573397616203502e-05
        rotation -0.9999982113395169 0.00047275268613733115 -0.0018313444964861228 1.2425810202889571
        children [
          DEF WHEEL Transform {
            translation 0.007 0 -0.05
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.1 0.1 0.1
                  transparency 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
            rotationStep 0.25
          }
        ]
        name "left wheel(1)"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    DEF front_ds DistanceSensor {
      translation 0 0 -0.045
      rotation 0 1 0 1.5708
      name "front_ds"
      recognitionColors [
        0 0 0
      ]
      lookupTable [
        0 0 0
        1.27 1.27 0
      ]
    }
    DEF right_ds DistanceSensor {
      translation 0.035 -0.01 -0.01
      name "right_ds"
      lookupTable [
        0 0 0
        1.27 1.27 0
      ]
    }
    DEF left_ds DistanceSensor {
      translation -0.035 -0.01 -0.01
      rotation 0 1 0 3.14159
      name "left_ds"
      lookupTable [
        0 0 0
        1.27 1.27 0
      ]
    }
    Camera {
      translation 0 -0.01 -0.045
      name "camera1"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
  ]
}
